_rules/SC2-4-6-descriptive-headings.md

```diff
@@ -0,0 +1,170 @@
+---
+name: Heading is descriptive
+test_type: atomic
+
+description: |
+   This rule checks that headings describe the topic or purpose of the content.
+   
+success_criterion:
+- 2.4.6 # Headings and labels
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Dagfinn Rømen
+- Geir Sindre Fossøy
+- Carlos Duarte
+---
+
+## Test procedure
+
+### Applicability
+
+This rule applies to any element with the [semantic role](#semantic-role) of heading that is either [visible](#visible) or [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+**Note**: This rule only applies to elements with the [semantic role](#semantic-role) of heading. Thus, it is a partial check for WCAG 2.0 success criterion 2.4.6, which applies to all headings. "Heading" is used in its general sense and includes titles and other ways to add a heading to different types of content.
+
+### Expectation
+
+Each target element describes the topic or purpose of the subsequent section.
+
+**Note**: Heading content defines the header of a section (whether explicitly marked up using sectioning content elements, or implied by the heading content itself). A section is a self-contained portion of written content that deals with one or more related topics or thoughts. A section may consist of one or more paragraphs and include graphics, tables, lists and sub-sections.
+
+**Note**: Headings do not need to be lengthy. A word, or even a single character, may suffice.
+
+## Assumptions
+_There are currently no assumptions._
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.4.6: Headings and Labels](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html) 
+- [G130: Providing descriptive headings](https://www.w3.org/WAI/WCAG21/Techniques/general/G130)
+- [H42: Using h1-h6 to identify headings](https://www.w3.org/WAI/WCAG21/Techniques/html/H42)
+- [ARIA12: Using role=heading to identify headings](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA12)
+- [HTML 5.2 Standard - Heading content](https://www.w3.org/TR/html52/dom.html#heading-content)
+
+## Test Cases
+
+### Passed
+
+#### Passed example 1
+
+Heading marked up with h-element that describes the topic or purpose of the following section of content.
+
+```html
+<h1 class="target">Opening Hours</h1>
+<p>We are open Monday through Friday from 10 to 16</p>
+```
+
+#### Passed example 2
+
+Heading marked up with `role="heading"` that describes the topic or purpose of the following section of content.
+
+```html
+<span role="heading">Opening Hours</span>
```

Comment:
and the level of the heading? The level is an important semantic info.
```html
<span role="heading" level="1">Opening Hours</span>
```

Reply:
Good point :)